### ISSUE TITLE
MythTV 34 Floating point exception

### ISSUE THREAD

**Issue by hallonstedt** [MythTV/mythtv#865]
* **Platform**:
Debian 12.4, 6.1.69 kernel (manually compiled), 8 cores, 8GB Ram (VM under Proxmox), Digital Devices dual DVB-S2 card.
MariaDB version: 10.1.37

* **MythTV version**:
mythbackend version: fixes/34 [v34.0-5c9cf67a497]

* **Component**:
mythbackend

### What steps will reproduce the bug?
Starting mythbackend and waiting 1-2 minutes

When I start the backend, it runs for 1-2 minutes and then it crashes with the same error message (see below)

I have compiled MythTV v34 on a brand new Debian since my old server, running MythTV v31 is running out of support. I tried to compile both the latest git version as well as the tar-file on the mythtv website. I also tried both traditional ./configure and the new cmake. Both compile with no errors and both results in a backend that constantly crashes.

I have used the ansible script to ensure dependencies are met. 

The only code-change I have made is adding `--break-system-packages` to bindings/python/Makefile PIP_OPTRIONS or it will refuse to install them.

### How often does it reproduce? Is there a required condition?
Every time

### What is the expected behaviour?
Not a crash!

Here are the last three rows from four different attempts to start with the command:
 /usr/local/bin/mythbackend --loglevel debug --logpath /var/log/mythtv
```
2024-02-14 16:49:04.848021 D [1758/1758] CoreContext housekeeper.cpp:117 (CheckRun) - ThemeUpdateNotifications Running? No/In window? No.
2024-02-14 16:49:04.848026 D [1758/1758] CoreContext housekeeper.cpp:117 (CheckRun) - UpdateRadioStreams Running? No/In window? No.
2024-02-14 16:49:22.035261 C [1758/1758] CoreContext signalhandling.cpp:292 (handleSignal) - Received Floating point exception: Code 1, PID 1594974737, UID 32540, Value 0x00000000

2024-02-14 16:49:04.848021 D [1758/1758] CoreContext housekeeper.cpp:117 (CheckRun) - ThemeUpdateNotifications Running? No/In window? No.
2024-02-14 16:49:04.848026 D [1758/1758] CoreContext housekeeper.cpp:117 (CheckRun) - UpdateRadioStreams Running? No/In window? No.
2024-02-14 16:49:22.035261 C [1758/1758] CoreContext signalhandling.cpp:292 (handleSignal) - Received Floating point exception: Code 1, PID 1594974737, UID 32540, Value 0x00000000

2024-02-14 16:51:48.774284 D [2250/2283] SSDP portchecker.cpp:193 (checkPort) - PortChecker::checkPort(): socket state 3
2024-02-14 16:51:53.647874 D [2250/2280] Expire portchecker.cpp:75 (checkPort) - PortChecker::checkPort(): host freeport port 3306 timeLimit 30000 linkLocalOnly 1
2024-02-14 16:52:32.684932 C [2250/2250] CoreContext signalhandling.cpp:292 (handleSignal) - Received Floating point exception: Code 1, PID 156328465, UID 32553, Value 0x00000000

2024-02-14 16:53:29.321677 D [2474/2507] SSDP configuration.cpp:39 (Load) - Loading /root/.mythtv/config.xml
2024-02-14 16:53:29.321782 D [2474/2507] SSDP configuration.cpp:203 (GetValue) - Using default for "UPnP/SSDP/MaxAge"
2024-02-14 16:54:01.186829 C [2474/2474] CoreContext signalhandling.cpp:292 (handleSignal) - Received Floating point exception: Code 1, PID 2077319697, UID 32689, Value 0x00000000
```

### Additional information

Any help with what I can try to identify the issue would be appreciated.

**Comment by kmdewaal** [comment]
Try running with gdb and show the callstack (with gdb command "bt" or "bt full") after it crashed.
You get better information when you have done a debug build. You can do that by adding "--compile-type=debug" to the configure options.

**Comment by hallonstedt** [comment]
I'll recompile with debug in the mean-time, I ran with gdb. Is this helpful?
```
2024-02-14 18:01:52.115500 D  Using default for "UPnP/SSDP/MaxAge"
2024-02-14 18:01:56.121312 D  PortChecker::checkPort(): host freeport port 3306 timeLimit 30000 linkLocalOnly 1
[Thread 0x7fff12ffd6c0 (LWP 11031) exited]
[Thread 0x7fff0b7fe6c0 (LWP 11029) exited]
[Thread 0x7fff13fff6c0 (LWP 11028) exited]
[Thread 0x7fffe6bfd6c0 (LWP 10992) exited]
[Thread 0x7fff137fe6c0 (LWP 11030) exited]
2024-02-14 18:02:36.847019 D  Running HouseKeeper.
2024-02-14 18:02:36.847031 D  DBCleanup Running? No/In window? No.
2024-02-14 18:02:36.847044 D  HardwareProfiler Running? No/In window? No.
2024-02-14 18:02:36.847046 D  JobQueueRecover Running? No/In window? No.
2024-02-14 18:02:36.848150 D  MythFillDatabase is disabled. Cannot run.
2024-02-14 18:02:36.848152 D  MythFillDB Running? No/In window? No.
2024-02-14 18:02:36.848155 D  RecordedArtworkUpdate Running? No/In window? No.
2024-02-14 18:02:36.848157 D  ThemeUpdateNotifications Running? No/In window? No.
2024-02-14 18:02:36.848162 D  UpdateRadioStreams Running? No/In window? No.

Thread 6 "TVRecEvent" received signal SIGFPE, Arithmetic exception.
[Switching to Thread 0x7fffe5b0f6c0 (LWP 11004)]
0x00007ffff7b16211 in EITScanner::StartActiveScan (this=0x7fffd4004360, rec=<optimized out>, max_seconds_per_multiplex=...) at eitscanner.cpp:298
298             auto nextChanIndex = MythRandom() % m_activeScanChannels.size();
(gdb) bt
#0  0x00007ffff7b16211 in EITScanner::StartActiveScan(TVRec*, std::chrono::duration<long, std::ratio<1l, 1l> >) (this=0x7fffd4004360, rec=<optimized out>, max_seconds_per_multiplex=...) at eitscanner.cpp:298
#1  0x00007ffff7b6732a in TVRec::run() (this=0x555555a565d0) at tv_rec.cpp:1561
#2  0x00007ffff6e91fde in MThread::run() (this=0x555555a58730) at mthread.cpp:319
#3  MThreadInternal::run() (this=<optimized out>) at mthread.cpp:78
#4  0x00007ffff58cbd43 in  () at /lib/x86_64-linux-gnu/libQt5Core.so.5
#5  0x00007ffff56a8134 in start_thread (arg=<optimized out>) at ./nptl/pthread_create.c:442
#6  0x00007ffff57287dc in clone3 () at ../sysdeps/unix/sysv/linux/x86_64/clone3.S:81
```

If you're really bored, here is the full stack;
```
(gdb) bt full
#0  0x00007ffff7b16211 in EITScanner::StartActiveScan(TVRec*, std::chrono::duration<long, std::ratio<1l, 1l> >) (this=0x7fffd4004360, rec=<optimized out>, max_seconds_per_multiplex=...) at eitscanner.cpp:298
        query = {<QSqlQuery> = {d = 0x7fffd40034e0}, m_db = 0x7fffd4003240, m_isConnected = true, m_returnConnection = true, m_lastPreparedQuery = {d = 0x7fffd40139a0}}
        nextChanIndex = <optimized out>
        sourceid = <optimized out>
        sourcename = {d = 0x7ffff5b97b20 <QArrayData::shared_null>}
        locker = {val = 140736750175089}
        __FUNCTION__ = "StartActiveScan"
#1  0x00007ffff7b6732a in TVRec::run() (this=0x555555a565d0) at tv_rec.cpp:1561
        allow_eit = <optimized out>
        inputids = std::vector of length 3, capacity 4 = {2, 5, 6}
        busy_input = {_vptr.InputInfo = 0x7ffff7f12690 <vtable for InputInfo+16>, m_name = {d = 0x7ffff5b97b20 <QArrayData::shared_null>}, m_sourceId = 0, m_inputId = 0, m_mplexId = 0, m_chanId = 0, m_displayName = {d = 0x7ffff5b97b20 <QArrayData::shared_null>}, m_recPriority = 0, m_scheduleOrder = 0, m_liveTvOrder = 0, m_quickTune = false}
        recEnd = {d = {d = 0x1, data = {status = 1, msecs = 0}}}
        conflicting_input = false
        lock = {val = 93824997484329}
        __FUNCTION__ = "run"
#2  0x00007ffff6e91fde in MThread::run() (this=0x555555a58730) at mthread.cpp:319
#3  MThreadInternal::run() (this=<optimized out>) at mthread.cpp:78
#4  0x00007ffff58cbd43 in  () at /lib/x86_64-linux-gnu/libQt5Core.so.5
#5  0x00007ffff56a8134 in start_thread (arg=<optimized out>) at ./nptl/pthread_create.c:442
        ret = <optimized out>
        pd = <optimized out>
        unwind_buf = {cancel_jmp_buf = {{jmp_buf = {140737046968000, 8869764650774470512, -416, 11, 140737488345744, 140737038577664, -8869715679134441616, -8869750266936123536}, mask_was_saved = 0}}, priv = {pad = {0x0, 0x0, 0x0, 0x0}, data = {prev = 0x0, cleanup = 0x0, canceltype = 0}}}
        not_first_call = <optimized out>
#6  0x00007ffff57287dc in clone3 () at ../sysdeps/unix/sysv/linux/x86_64/clone3.S:81

```

**Comment by kmdewaal** [comment]
This does look familiar, it is code that I did modify recently.
Please do generate the tracebacks (yes, also the full) with the debug build.

**Comment by kmdewaal** [comment]
I did just push a fix to master/v35-pre and v34 in commits 36d76494c1222311d49c9fa84312c69ef2526c45 and b6aabcbbf3f2fcc1bc5758f4e8943018c878b2fc
Please test this and when you start mythbackend please do add "-v eit". This should give an EIT message about active scan like this:
`2024-02-14 19:01:42.919142 I  EITScanner[42]: EIT scan channel table for source 12 'Astra-2' with 36 multiplexes`
but then with 0 multiplexes if my understanding of the problem is correct.

**Comment by hallonstedt** [comment]
You are right. And now it has been running for a whooping 20 minutes with no crash. Thank you!
```
2024-02-14 21:21:57.062653 I [1223/1239] TVRecEvent tv_rec.cpp:1518 (run) - TVRec[1]: EIT scanning enabled for input 1 connected to video source 1 'Canal Digital'
2024-02-14 21:21:57.063124 I [1223/1239] TVRecEvent tv_rec.cpp:1558 (run) - TVRec[1]: Start EIT active scan on input 1
2024-02-14 21:21:57.064146 I [1223/1239] TVRecEvent eitscanner.cpp:306 (StartActiveScan) - EITScanner[1]: EIT scan channel table for source 1 'Canal Digital' with 0 multiplexes

```
